Index: foma/regex.y
===================================================================
--- foma.orig/regex.y
+++ foma/regex.y
@@ -385,7 +385,7 @@ network12: fend    { $$ = $1; } |
          ADDSINK network RPAREN { $$ = fsm_add_sink($2,1); } |
          LEFTREWR network COMMA network RPAREN { $$ = fsm_left_rewr($2,$4); } |
          FLATTEN network COMMA network RPAREN { $$ = fsm_flatten($2,$4); } |
-         SUBLABEL network COMMA network COMMA network RPAREN { $$ = fsm_substitute_label($2, fsm_network_to_char($4), $6) } |
+         SUBLABEL network COMMA network COMMA network RPAREN { $$ = fsm_substitute_label($2, fsm_network_to_char($4), $6); } |
          CLOSESIGMA    network RPAREN { $$ = fsm_close_sigma(fsm_copy($2), 0); } |
          CLOSESIGMAUNK network RPAREN { $$ = fsm_close_sigma(fsm_copy($2), 1); } |
          EQSUBSTRINGS network COMMA network COMMA network RPAREN { $$ = fsm_equal_substrings($2,$4,$6); }
