Description: Do not timestamp in output.
Index: b/main-cicling.cc
===================================================================
--- a/main-cicling.cc	(revision 4596)
+++ b/main-cicling.cc	(working copy)
@@ -54,7 +54,6 @@
     std::cerr <<
     "\n" <<
     PACKAGE_STRING << std::endl <<
-    __DATE__ << " " __TIME__ << std::endl <<
     "copyright (C) 2009 University of Helsinki\n";
     return true;
 }
Index: b/main-fsmnlp-2012.cc
===================================================================
--- a/main-fsmnlp-2012.cc	(revision 4596)
+++ b/main-fsmnlp-2012.cc	(working copy)
@@ -73,7 +73,6 @@
     std::cerr <<
     "\n" <<
     PACKAGE_STRING << std::endl <<
-    __DATE__ << " " __TIME__ << std::endl <<
     "copyright (C) 2009 - 2011 University of Helsinki\n";
     return true;
 }
Index: b/main-norvig.cc
===================================================================
--- a/main-norvig.cc	(revision 4596)
+++ b/main-norvig.cc	(working copy)
@@ -54,7 +54,6 @@
     std::cerr <<
     "\n" <<
     PACKAGE_STRING << std::endl <<
-    __DATE__ << " " __TIME__ << std::endl <<
     "copyright (C) 2009 University of Helsinki\n";
     return true;
 }
Index: b/main-survey.cc
===================================================================
--- a/main-survey.cc	(revision 4596)
+++ b/main-survey.cc	(working copy)
@@ -95,7 +95,6 @@
     std::cerr <<
     "\n" <<
     PACKAGE_STRING << std::endl <<
-    __DATE__ << " " __TIME__ << std::endl <<
     "copyright (C) 2009 - 2011 University of Helsinki\n";
     return true;
 }
Index: b/main.cc
===================================================================
--- a/main.cc	(revision 4596)
+++ b/main.cc	(working copy)
@@ -151,7 +151,6 @@
     std::cout <<
     "\n" <<
     PACKAGE_STRING << std::endl <<
-    __DATE__ << " " __TIME__ << std::endl <<
     "copyright (C) 2009 - 2016 University of Helsinki\n";
     return true;
 }
