diff --git a/Makefile.am b/Makefile.am
index 9423c5c..f097d43 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -122,7 +122,7 @@ TESTS=tests/basic-zhfst.sh tests/basic-edit1.sh \
 	  tests/empty-descriptions.sh tests/empty-titles.sh tests/empty-locale.sh \
 	  tests/trailing-spaces.sh tests/bad-errormodel.sh tests/empty-zhfst.sh \
 	  tests/analyse-spell.sh tests/no-errormodel.sh
-XFAIL_TESTS=tests/empty-descriptions.sh tests/empty-titles.sh tests/empty-locale.sh tests/empty-zhfst.sh
+XFAIL_TESTS=tests/empty-zhfst.sh
 
 if CAN_DOXYGEN
 doxygen:
