diff --git a/tools/src/hfst-optimized-lookup.cc b/tools/src/hfst-optimized-lookup.cc
index a31e8a4..3d56389 100644
--- a/tools/src/hfst-optimized-lookup.cc
+++ b/tools/src/hfst-optimized-lookup.cc
@@ -97,7 +97,6 @@ bool print_version(void)
   std::cout <<
     "\n" <<
     PACKAGE_STRING << std::endl <<
-    __DATE__ << " " __TIME__ << std::endl <<
     "copyright (C) 2009 University of Helsinki\n";
   return true;
 }
diff --git a/tools/src/hfst-proc/hfst-proc.cc b/tools/src/hfst-proc/hfst-proc.cc
index c0d293a..bd4d930 100644
--- a/tools/src/hfst-proc/hfst-proc.cc
+++ b/tools/src/hfst-proc/hfst-proc.cc
@@ -154,7 +154,6 @@ bool print_version(void)
     " (" << PACKAGE_STRING << ")" <<
 #endif
     std::endl <<
-    __DATE__ << " " __TIME__ << std::endl <<
     "copyright (C) 2009-2011 University of Helsinki\n";
   return true;
 }
