Index: configure.ac
===================================================================
--- a/configure.ac	(revision 4204)
+++ b/configure.ac	(working copy)
@@ -721,6 +721,7 @@
 AC_CHECK_DECLS([rl_completion_suppress_append, rl_completion_matches])
 AC_CHECK_FUNCS([floor strchr strdup strerror strncasecmp strcspn strtol strtoul error error_at_line strndup getline getdelim getopt_long strtod xstrdup set_program_name setprogname setlocale nl_langinfo])
 
+AM_CONDITIONAL([WINDOWS], [test x$version_type = xwindows])
 
 # Checks for system services
 
Index: libhfst/src/Makefile.am
===================================================================
--- a/libhfst/src/Makefile.am	(revision 4204)
+++ b/libhfst/src/Makefile.am	(working copy)
@@ -60,6 +60,10 @@
 #  libhfst_la_LIBADD += -lxfsm
 #endif
 
+if WINDOWS
+  libhfst_la_LIBADD += -ldl
+endif
+
 # we only have one library target but would like a particular includedir
 # structure for various headers, so we do this
 hfstincludedir = $(includedir)/hfst
Index: tools/src/hfst-commandline.cc
===================================================================
--- a/tools/src/hfst-commandline.cc	(revision 4204)
+++ b/tools/src/hfst-commandline.cc	(working copy)
@@ -502,7 +502,7 @@
 int
 hfst_mkstemp(char* templ)
 {
-#ifdef WINDOWS
+#ifdef _WIN32
   error(EXIT_FAILURE, errno, 
         "'int hfst_mkstemp(char * temp1)' not implemented for windows");
 #else 
Index: tools/src/parsers/XfstCompiler.cc
===================================================================
--- a/tools/src/parsers/XfstCompiler.cc	(revision 4204)
+++ b/tools/src/parsers/XfstCompiler.cc	(working copy)
@@ -33,7 +33,7 @@
 
 #include <cstdarg>
 
-#ifndef WINDOWS
+#ifndef _WIN32
   #include <glob.h>
 #else
   #include <windows.h>
@@ -2493,7 +2493,7 @@
   XfstCompiler& 
   XfstCompiler::print_dir(const char* globdata, FILE* outfile)
     {
-#ifndef WINDOWS
+#ifndef _WIN32
       glob_t globbuf;
 
       int rv = glob(globdata, 0, NULL, &globbuf);
