diff --git a/configure.ac b/configure.ac
index 9e326e9..e1c1d10 100644
--- a/configure.ac
+++ b/configure.ac
@@ -178,7 +178,8 @@ AC_RUN_IFELSE([AC_LANG_PROGRAM([
               [AC_MSG_FAILURE(m4_normalize([
                    "Checking for float equality... no
                    Compile with -msse -mfpmath=sse if using g++."
-              ]))])
+              ]))],
+              [echo "Skipping float equality test for cross compilation"])
 
 AC_CHECK_LIB([dl], dlopen, [DL_LIBS=-ldl])
 AC_SUBST([DL_LIBS])
