Description: Enable C++17, needed for dependent packages
diff --git a/configure.ac b/configure.ac
index ae727b1..4d597de 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5,7 +5,7 @@ AM_PROG_AR
 AC_PROG_CXX
 # This library does not throw exceptions, so we do not generate exception
 # handling code. However, users are free to re-enable exception handling.
-CXX="$CXX -std=c++11 -fno-exceptions"
+CXX="$CXX -std=c++17 -fno-exceptions"
 
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
